rascal2/dialogs/project_dialog.py

```diff
@@ -0,0 +1,209 @@
+
+        self.cancel_button = QtWidgets.QPushButton(" Cancel", self)
+        self.cancel_button.setIcon(QtGui.QIcon(path_for("cancel.png")))
+        self.cancel_button.clicked.connect(self.cancel_project_creation)
+        self.cancel_button.setStyleSheet(self._button_style.format("#E34234"))
+
+    def cancel_project_creation(self) -> None:
+        """
+        Cancel project creation.
+        """
+        self.close() if self.parent().new_project_action_called else self.parent().toggleView()
+        self.reset_variables()
+
+    def reset_variables(self) -> None:
+        """
+        Reset variables.
+        """
+        self.project_folder.setText("")
+        self.project_name.setText("")
+        self.name_error = False
+        self.folder_error = False
+
+    def open_folder_selector(self) -> None:
+        """
+        Open folder selector.
+        """
+        folder_path = QtWidgets.QFileDialog.getExistingDirectory(self, "Select Folder")
+        if folder_path:
+            files_in_folder = list(filter(lambda x: x[0] != ".", os.listdir(folder_path)))
+            if files_in_folder:
+                self.handle_error("folder")
+            else:
+                self.handle_success("folder")
+                self.project_folder.setText(folder_path)
+
+    def handle_error(self, tag: Literal["name", "folder"]) -> None:
+        """
+        Display error msgs.
+
+        Parameters
+        ----------
+        tag: Literal["name", "folder"]
+            Specifies whether to show the error
+            for the project name or folder.
+        """
+        getattr(self, "project_" + tag).setStyleSheet(self._line_edit_error_style)
+        getattr(self, "project_" + tag + "_error").show()
+        setattr(self, tag + "_error", True)
+
+    def handle_success(self, tag: Literal["name", "folder"]) -> None:
+        """
+        Hide error msgs.
+
+        Parameters
+        ----------
+        tag: Literal["name", "folder"]
+            Specifies whether to clear the error
+            for the project name or folder.
+        """
+        getattr(self, "project_" + tag).setStyleSheet("")
+        getattr(self, "project_" + tag + "_error").hide()
+        setattr(self, tag + "_error", False)
+
+    def verify_inputs(self, tag: Literal["name", "folder"]) -> None:
+        """
+        Verify inputs.
+
+        Parameters
+        ----------
+        tag: Literal["name", "folder"]
+            Specifies whether the input verification
+            is for the project name or folder.
+        """
+        if getattr(self, "project_" + tag).text() == "":
+            self.handle_error(tag)
+        else:
+            self.handle_success(tag)
+
+    def create_project(self) -> None:
+        """
+        Create project if inputs verified.
+        """
+        self.verify_inputs("name")
+        self.verify_inputs("folder")
+        if not self.name_error and not self.folder_error:
```

Comment:
`self.name_error` and `self.folder_error` are only read or written (apart from in `reset_variables`) within this function. would it be simpler if we just made `verify_inputs` return a bool for whether verification succeeded? then the class would be simpler, `reset_variables` wouldn't need to worry about resetting the attributes, and this function could just be

```python
def create_project(self):
    if self.verify_inputs("name") and self.verify_inputs("folder"):
        self.parent().createNewProject()
```